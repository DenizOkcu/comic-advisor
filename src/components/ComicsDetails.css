.details {
  background-color: white;
  border-radius: 2px;
  border: 1px solid #bbb;
  box-shadow: 2px 2px 5px #eee;
  cursor: pointer;
  display: inline-block;
  height: 99vh;
  left: 0px;
  overflow-y: auto;
  padding: 0 10px 10px 10px;
  position: fixed;
  top: 0px;
  z-index: 10;
  transition: transform 800ms ease-in-out;
}

/* animates the details view to slide in... */
.details.hidden-on-mobile {
  transform: translate(-600px);
  display: inline-block;
}
/* ... and out */
.details.visible-on-mobile {
  transform: translate(0px);
  display: inline-block;
}

.close-details {
  background-color: red;
  border: 1px solid black;
  color: white;
  cursor: pointer;
  font-family: "Bangers", Arial Black;
  font-size: 30px;
  font-weight: bold;
  height: 44px;
  position: fixed;
  right: -13px;
  text-align: center;
  top: 50%;
  transform: rotate(-90deg);
  vertical-align: middle;
  width: 92px;
}

.hidden-on-mobile {
  display: none;
}

.detail-text {
  flex: 1;
  padding: 10px 0 0;
}

/* Larger than mobile */
@media (min-width: 550px) {
  .details {
    border-radius: 2px;
    border: 1px solid #bbb;
    box-shadow: 2px 2px 5px #eee;
    cursor: pointer;
    display: inline-block;
    flex: 1;
    height: 100%;
    padding: 0 10px 10px 10px;
    position: relative;
  }

  /* keine animation intablet und desktop */
  .details.hidden-on-mobile,
  .details.visible-on-mobile {
    transform: translate(0px);
    display: inline-block;
  }

  .hidden-on-mobile {
    display: inline-block;
  }

  .hidden-on-gt-mobile {
    display: none;
  }

  .detail-text {
    flex: 1;
    padding: 10px 0 0;
  }
}

/* Larger than tablet */
@media (min-width: 850px) {
  .details {
    border-radius: 2px;
    border: 1px solid #bbb;
    box-shadow: 2px 2px 5px #eee;
    cursor: pointer;
    display: inline-block;
    flex: 2;
    padding: 0 10px 10px 10px;
  }

  .detail-cover-text-container {
    display: flex;
  }

  .detail-cover {
    flex: 1;
    border-radius: 2px;
    margin-bottom: 15px;
  }

  .detail-text {
    flex: 1;
    padding: 0 0 0 13px;
  }
}

.detail-title {
  margin: 3px;
  font-size: 1.5em;
  font-family: "Bangers", Arial Black;
  letter-spacing: 0.25rem;
}

.primary-button {
  background-color: red;
  font-family: "Bangers", Arial Black;
  color: white;
  padding: 0.05em 0;
  text-decoration: none;
  text-transform: uppercase;
  text-align: center;
  font-size: 2em;
  border-radius: 5px;
  display: inline-block;
  width: 100%;
  margin-top: 10px;
}
