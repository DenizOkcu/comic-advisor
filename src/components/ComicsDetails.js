import React from "react";
import { Row } from "./grid";
import ComicDetailsCreators from "./ComicDetailsCreators";

const ComicsDetails = ({ comic }) => {
  // zeile 18+19 hier bearbeiten
  // oder sogar thumbnail komponente anlegen

  // comic image komponente bauen
  // platzhalter rndern fürs laden
  // size übergeben

  return (
    <main>
      {comic ? (
        <article className="eight columns hide-on-small">
          <h4>Comics Details</h4>
          <hr />
          <Row>
            <div className="twelve columns">
              <h5 className="detail-title">{comic.title}</h5>
            </div>
          </Row>
          <Row>
            <div className="five columns">
              <img
                src={comic.coverPath}
                className="detail-img"
                alt={comic.title}
              />
            </div>
            <div className="seven columns">
              <ComicDetailsCreators creators={comic.creators} />
            </div>
          </Row>
        </article>
      ) : (
        ""
      )}
    </main>
  );
};

export default ComicsDetails;
