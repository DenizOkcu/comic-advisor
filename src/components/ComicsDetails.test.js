import React from "react";
import { shallow } from "enzyme";
import ComicsDetails from "./ComicsDetails";

describe("ComicsDetails", () => {
  it('should render correctly in "debug" mode', () => {
    const comic = {
      id: 1,
      creators: { items: [{ name: "writer name" }] },
      thumbnail: { path: "test", extension: "jpg" },
      prices: [{ price: 3.99 }],
      title: "title",
      pageCount: 32
    };
    const component = shallow(<ComicsDetails comic={comic} debug />);

    expect(component).toMatchSnapshot();
  });
});
