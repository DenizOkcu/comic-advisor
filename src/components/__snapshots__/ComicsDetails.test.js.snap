// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComicsDetails should render correctly in "debug" mode 1`] = `
<main>
  <article
    className="eight columns hide-on-small"
  >
    <h4>
      Comics Details
    </h4>
    <hr />
    <Row>
      <div
        className="twelve columns"
      >
        <h5
          className="detail-title"
        >
          title
        </h5>
      </div>
    </Row>
    <Row>
      <div
        className="five columns"
      >
        <img
          alt="title"
          className="detail-img"
        />
      </div>
      <div
        className="seven columns"
      >
        <ComicDetailsCreators
          creators={
            Array [
              Object {
                "name": "writer name",
              },
            ]
          }
        />
      </div>
    </Row>
  </article>
</main>
`;
